### PR TITLE
Fix a bug of build number.

### LINKED ISSUE #29: Build fails if package is already present in the anaconda channel
Not sure what's happening b/c if it was already present it shouldn't try building it either.

```
2015-11-05 03:05:26,938 INFO: Uploading: anaconda -t {token} upload -u {channel} {built}

Using anaconda-server api site https://api.anaconda.org

detecting package type ...

conda

extracting package attributes for upload ...

done

Uploading file biocore/minced/0.2.0/osx-64/minced-0.2.0-1.tar.bz2 ... 

Distribution already exists. Please use the -i/--interactive or --force options or `anaconda remove biocore/minced/0.2.0/osx-64/minced-0.2.0-1.tar.bz2`

[Conflict] (u'file osx-64/minced-0.2.0-1.tar.bz2 already exists for package minced version 0.2.0', 409)

Traceback (most recent call last):

  File "ci/main.py", line 133, in <module>

    build_upload_recipes(sys.argv[1], sys.argv[2])

  File "ci/main.py", line 50, in build_upload_recipes

    upload(name, version, channel)

  File "ci/main.py", line 129, in upload

    shell=True)

  File "/Users/travis/miniconda/lib/python2.7/subprocess.py", line 540, in check_call

    raise CalledProcessError(retcode, cmd)

subprocess.CalledProcessError: Command 'anaconda -t ze-b2e094e1-af48-4b13-a678-ac6f0e5e9581 upload -u biocore /Users/travis/miniconda/conda-bld/osx-64/minced-0.2.0-1.tar.bz2' returned non-zero exit status 1

The command "python ci/main.py recipes biocore" exited with 1.
```

### PR DESCRIPTION
**Pull request by RNAer**
Find max build number in the remote channel. Fix #29 and #30

### CODE REVIEW

**Comment by RNAer** [comment]
reverse the logic of upload check, for it was getting a little unwieldy to add layers of logic of checking both version and build number together.

File: ci/main.py [hunk 1]
```diff
@@ -118,14 +126,11 @@ def upload(name, version, channel):
...
     built = glob.glob(built_glob)[0]
-    upload_cmd = 'anaconda -t {token} upload -u {channel} {built}'
+    upload_cmd = 'anaconda -t {token} upload -u %s %s' % (channel, built)
     # Do not show decrypted token!
     log.info('Uploading: {0}'.format(upload_cmd))
     check_call(
-        upload_cmd.format(
-            token=os.environ['ANACONDA_TOKEN'],
-            built=built,
-            channel=channel),
```
Comment (Jorge-C) [inline ci/main.py:128]: Can you keep the command formatting in the same call? Thanks!

Comment (RNAer) [inline ci/main.py:128]: I changed it conveniently while I was at it. you think it is gilding the
lily?

Comment (Jorge-C) [inline ci/main.py:128]: I agree that making that change was gilding the lily (cool expression btw) and that's why I was asking to undo, but it's not a big deal and will merge nevertheless.

**Pull request merged**